lib/Dialect/FIRRTL/Import/FIRAnnotations.cpp

```diff
@@ -137,15 +137,17 @@ buildNLA(CircuitOp circuit, size_t nlaSuffix,
   OpBuilder b(circuit.getBodyRegion());
   MLIRContext *ctxt = circuit.getContext();
   SmallVector<Attribute> insts;
-  for (auto &nla : nlas) {
+  for (size_t i = 0, e = nlas.size(); i < e; ++i) {
+    auto &nla = nlas[i];
     // Assumption: Symbol name = Operation name.
     auto module = std::get<1>(nla);
-    auto inst = std::get<2>(nla);
-    if (inst.empty())
+    if (i == e - 1)
       insts.push_back(FlatSymbolRefAttr::get(ctxt, module));
-    else
+    else {
+      auto inst = std::get<2>(nla);
       insts.push_back(hw::InnerRefAttr::get(StringAttr::get(ctxt, module),
                                             StringAttr::get(ctxt, inst)));
+    }
   }
```

Comment:
Can something like this work?
```
  for (size_t i = 0, e = nlas.size() - 1; i < e; ++i) {
    auto &nla = nlas[i];
    // Assumption: Symbol name = Operation name.
    auto module = std::get<1>(nla);
    auto inst = std::get<2>(nla);
    insts.push_back(hw::InnerRefAttr::get(StringAttr::get(ctxt, module),
                                          StringAttr::get(ctxt, inst)));
  }
  insts.push_back(FlatSymbolRefAttr::get(ctxt, module));
 ```

Reply:
Yeah, this looks better. 👍

Reply:
I'd be careful about `size_t... = size() - 1` re:wraparound, although looks like `nlas` being non-empty is already a precondition.  But when that goes wrong it's not always as easy to debug as other sorts of failures.

Anyway, since the index is no longer needed, consider replacing the `SmallVectorImpl<> &` with an `ArrayRef<>` (`MutableArrayRef` if this was written to) and you can simplify a bit more if you want:

```c++
static FlatSymbolRefAttr
buildNLA(CircuitOp circuit, size_t nlaSuffix,
         ArrayRef<std::tuple<std::string, StringRef, StringRef>> nlas) {
  OpBuilder b(circuit.getBodyRegion());
  MLIRContext *ctxt = circuit.getContext();
  SmallVector<Attribute> insts;
  for (auto &[module, inst, _] : nlas.drop_back()) {
    // Assumption: Symbol name = Operation name.
    insts.push_back(hw::InnerRefAttr::get(StringAttr::get(ctxt, module),
                                          StringAttr::get(ctxt, inst)));
  }
  insts.push_back(FlatSymbolRefAttr::get(ctxt, std::get<1>(nlas.back())));

```